src/utils/EIP712.sol

```diff
@@ -36,7 +36,7 @@ contract EIP712 {
     /// or if the chain id changes due to a hard fork,
     /// the domain separator will be seamlessly calculated on-the-fly.
     constructor() {
-        _cachedThis = address(this);
+        _cachedThis = uint256(uint160(address(this)));
```

Comment:
what's the optimization gain here?

Reply:
Yeah also curious why we are doing this?

Reply:
tldr: these changes come from Solady. Since I do not import the project, I just updated the local version ^